exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricConfiguration.java

```diff
@@ -0,0 +1,98 @@
+package com.google.cloud.opentelemetry.metric;
+
+import com.google.auth.Credentials;
+import com.google.auto.value.AutoValue;
+import com.google.cloud.ServiceOptions;
+import com.google.cloud.monitoring.v3.MetricServiceClient;
+import com.google.cloud.monitoring.v3.stub.MetricServiceStub;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/** Configurations for {@link MetricExporter}. */
+@AutoValue
+@Immutable
+public abstract class MetricConfiguration {
+
+    private static final String DEFAULT_PROJECT_ID =
+            Strings.nullToEmpty(ServiceOptions.getDefaultProjectId());
+    private static final boolean DEFAULT_ADD_UNIQUE_IDENTIFIER = false;
+
+    MetricConfiguration() {}
+
+    /**
+     * Returns the {@link Credentials}.
+     *
+     * @return the {@code Credentials}.
+     */
+    @Nullable
+    public abstract Credentials getCredentials();
+
+    /**
+     * Returns the cloud project id.
+     *
+     * @return the cloud project id.
+     */
+    public abstract String getProjectId();
+
+    /**
+     * Returns the client.
+     *
+     * @return the client.
+     */
+    public abstract MetricServiceClient getClient();
+
+    /**
+     * Returns whether a unique identifier will be added.
+     *
+     * @return whether a unique identifier will be added.
+     */
+    public abstract boolean getAddUniqueIdentifier();
+
+    /**
+     * Returns a MetricsServiceStub instance used to make RPC calls.
+     *
+     * @return the metrics service stub.
+     */
+    @Nullable
+    public abstract MetricServiceStub getMetricServiceStub();
+
+    public static Builder builder() {
+        return new AutoValue_MetricConfiguration.Builder();
+    }
+
+    /** Builder for {@link MetricConfiguration}. */
+    @AutoValue.Builder
+    public abstract static class Builder {
+
+        Builder() {}
```

Comment:
You need to add following setters:
```java
    public abstract static class Builder {

        Builder() {}


        public abstract Builder setCredentials(Credentials credentials);

        public abstract Builder setAddUniqueIdentifier(boolean addUniqueIdentifier);

        public abstract Builder setMetricServiceStub(MetricServiceStub stub);
```
after making this change the compilation worked for me (the EndToEndTest fails, but that's probably unrelated).

Reply:
Thank you!